VCDA-395 and VCDA-396: Adding commands for listing and deleting org vdc networks

rocknes opened the pull request:
* Added commands to list and delete both direct networks and isolated org vdc networks.

Manual testing of commands
------
(vcd-cli) C:\code\vcd-cli\vcd_cli>vcd network direct list
name
[------]
dnet1
Dnet1

(vcd-cli) C:\code\vcd-cli\vcd_cli>vcd network direct delete dnet1
Are you sure you want to delete the OrgVdc Network? [y/N]: y
networkDelete: Deleting Network dnet1(3c3a916e-6601-4df5-84f0-804d2774757f)
task: 93732628-5b4f-45e2-ad10-9c8302b12e13, Deleted Network dnet1(3c3a916e-6601-4df5-84f0-804d2774757f), result: success

To test --force option, I attached the network to a vApp and tried to delete the network with and without the option --force
(vcd-cli) C:\code\vcd-cli\vcd_cli>vcd network direct delete Dnet1
Are you sure you want to delete the OrgVdc Network? [y/N]: y
Usage: vcd network direct delete [OPTIONS] <name>

Error: Status code: 400/BAD_REQUEST, [ 0d0a6cc5-d12b-4b81-a8be-183380c89113 ] Entity Dnet1 cannot be deleted, because it is in use. (request id: 0d0a6cc5-d12b-4b81-a8be-183380c89113)

(vcd-cli) C:\code\vcd-cli\vcd_cli>vcd network direct delete Dnet1 -f
Are you sure you want to delete the OrgVdc Network? [y/N]: y
networkDelete: Deleting Network Dnet1(77936cfb-58c9-44bb-b958-732f928f13e3)
task: e8013c54-4726-4e45-8ace-3cd18d2d35c4, Deleted Network Dnet1(77936cfb-58c9-44bb-b958-732f928f13e3), result: success


(vcd-cli) C:\code\vcd-cli\vcd_cli>vcd network isolated  list
name
[--------]
iso-net1

(vcd-cli) C:\code\vcd-cli\vcd_cli>vcd network isolated delete iso-net1
Are you sure you want to delete the OrgVdc Network? [y/N]: y
networkDelete: Deleting Network iso-net1(80980e57-7d23-4efb-b0d6-86aebfe10949)
task: a99eb56c-0a2a-4d96-8e47-7966fabe7ef2, Deleted Network iso-net1(80980e57-7d23-4efb-b0d6-86aebfe10949), result: success

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vmware/vcd-cli/173)
<!-- Reviewable:end -->

sahithi commented:
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[vcd_cli/network.py, line 75 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/173#-L3ZWei77VtDN_B9VsBX:-L3ZWei77VtDN_B9VsBY:byrs6st) ([raw file](https://github.com/vmware/vcd-cli/blob/5313486ad9b00a166d20c2e4cac5a1abcc1b3e59/vcd_cli/network.py#L75)):*
> ```Python
>             Create an org vdc network which is directly connected
>             to an external network.
>         vcd network direct list
> ```

New line between examples (at several places in the change-set)

---

*[vcd_cli/network.py, line 95 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/173#-L3ZaOKhmQFGRIprvJFi:-L3ZaOKhmQFGRIprvJFj:bfxpbjc) ([raw file](https://github.com/vmware/vcd-cli/blob/5313486ad9b00a166d20c2e4cac5a1abcc1b3e59/vcd_cli/network.py#L95)):*
> ```Python
>     Note
>         Both System Administrators and Organization Administrators can create,
>         delete or list isolated org vdc networks.
> ```

I like the note! I will incorporate similar note for my rights commands (only sys-admin can execute few).

---

*[vcd_cli/network.py, line 311 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/173#-L3Z__IPOUuNrKuZnwwy:-L3Z__IQ7vY_vHcbqvoB:b8es0nq) ([raw file](https://github.com/vmware/vcd-cli/blob/5313486ad9b00a166d20c2e4cac5a1abcc1b3e59/vcd_cli/network.py#L311)):*
> ```Python
>         result = []
>         for direct_net in direct_nets:
>             result.append({'name': direct_net.get('name')})
> ```

result.append(to_dict(direct_net, exclude('href'..))
This can add more fields like description etc.

---

*[vcd_cli/network.py, line 331 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/173#-L3Z_uQ1lFoR5EvXKqoc:-L3Z_uQ1lFoR5EvXKqod:b-6aw4vt) ([raw file](https://github.com/vmware/vcd-cli/blob/5313486ad9b00a166d20c2e4cac5a1abcc1b3e59/vcd_cli/network.py#L331)):*
> ```Python
>         result = []
>         for isolated_net in isolated_nets:
>             result.append({'name': isolated_net.get('name')})
> ```

same here use to_dict()

---

*[vcd_cli/network.py, line 362 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/173#-L3Za-ijwKqYlK8IC9-Y:-L3Za-ijwKqYlK8IC9-Z:b-b0e3hw) ([raw file](https://github.com/vmware/vcd-cli/blob/5313486ad9b00a166d20c2e4cac5a1abcc1b3e59/vcd_cli/network.py#L362)):*
> ```Python
>         vdc = VDC(client, href=in_use_vdc_href)
> 
>         result = vdc.delete_direct_orgvdc_network(name=name, force=force)
> ```

rename the variable to delete_task

---

*[vcd_cli/network.py, line 393 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/173#-L3Za5-ZIkkHZcJGCT4w:-L3Za5-ZIkkHZcJGCT4x:b-b0e3hw) ([raw file](https://github.com/vmware/vcd-cli/blob/5313486ad9b00a166d20c2e4cac5a1abcc1b3e59/vcd_cli/network.py#L393)):*
> ```Python
>         vdc = VDC(client, href=in_use_vdc_href)
> 
>         result = vdc.delete_isolated_orgvdc_network(name=name, force=force)
> ```

rename the variable to delete_task

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/173)*
<!-- Sent from Reviewable.io -->

hodgesrm commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/173#-:-L3ZkYlMGXd3yo3bPkbR:bnfp4nl)*
<!-- Sent from Reviewable.io -->

rocknes commented:
Review status: 0 of 1 files reviewed at latest revision, 5 unresolved discussions.

---

*[vcd_cli/network.py, line 75 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/173#-L3ZWei77VtDN_B9VsBX:-L3ZurzEMMbIl27-F712:bqmjwcz) ([raw file](https://github.com/vmware/vcd-cli/blob/5313486ad9b00a166d20c2e4cac5a1abcc1b3e59/vcd_cli/network.py#L75)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

New line between examples (at several places in the change-set)
</blockquote></details>

Added.

---

*[vcd_cli/network.py, line 311 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/173#-L3Z__IPOUuNrKuZnwwy:-L3ZuuK_Jh0CLopNVAeZ:b4806vw) ([raw file](https://github.com/vmware/vcd-cli/blob/5313486ad9b00a166d20c2e4cac5a1abcc1b3e59/vcd_cli/network.py#L311)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

result.append(to_dict(direct_net, exclude('href'..))
This can add more fields like description etc.
</blockquote></details>

I think description should be part of info command and not list, since description can  be multi-line and that will distort the output tabular format.

There is nothing apart from name of  network which makes sense to show as part of the list command. FenceMode is already taken care of since sub command structure clarifies what type of network we are dealing with. Configuration data is mostly ip scopes and gateway info which in my opinion should be part of info command.

---

*[vcd_cli/network.py, line 331 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/173#-L3Z_uQ1lFoR5EvXKqoc:-L3ZvQ43YhWlSjUByVIk:brmy0b2) ([raw file](https://github.com/vmware/vcd-cli/blob/5313486ad9b00a166d20c2e4cac5a1abcc1b3e59/vcd_cli/network.py#L331)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

same here use to_dict()
</blockquote></details>

Addressed in the previous comment

---

*[vcd_cli/network.py, line 362 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/173#-L3Za-ijwKqYlK8IC9-Y:-L3ZvS9WZ3_P7Gg_MqHF:bb6un1j) ([raw file](https://github.com/vmware/vcd-cli/blob/5313486ad9b00a166d20c2e4cac5a1abcc1b3e59/vcd_cli/network.py#L362)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

rename the variable to delete_task
</blockquote></details>

Done

---

*[vcd_cli/network.py, line 393 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/173#-L3Za5-ZIkkHZcJGCT4w:-L3Zvm5PGxSsF27D0QmE:bb6un1j) ([raw file](https://github.com/vmware/vcd-cli/blob/5313486ad9b00a166d20c2e4cac5a1abcc1b3e59/vcd_cli/network.py#L393)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

rename the variable to delete_task
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/173)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r1.
Review status: 0 of 1 files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/173#-:-L3Zyj36YNaneDKC0AXz:bnfp4nl)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/173)*
<!-- Sent from Reviewable.io -->
